tests/phpunit/tests/media.php

```diff
@@ -4339,6 +4390,130 @@ public function test_wp_get_loading_optimization_attributes( $context ) {
 		}
 	}
 
+	/**
+	 * Tests that wp_get_loading_optimization_attributes() does not return lazy loading attributes when arbitrary contexts are used before the main query loop.
+	 *
+	 * @ticket 58894
+	 *
+	 * @covers ::wp_get_loading_optimization_attributes
+	 *
+	 * @dataProvider data_wp_get_loading_optimization_attributes_arbitrary_contexts
+	 *
+	 * @param string $context
+	 */
+	public function test_wp_get_loading_optimization_attributes_with_arbitrary_contexts_before_main_query_loop( $context ) {
+		$attr = $this->get_width_height_for_high_priority();
+
+		$query = $this->get_new_wp_query_for_published_post();
+
+		while ( have_posts() ) {
+			the_post();
+
+			$this->assertSame(
+				array( 'loading' => 'lazy' ),
+				wp_get_loading_optimization_attributes( 'img', $attr, $context ),
+				'The "loading" attribute should be "lazy" while in the loop but not in the main query.'
+			);
+		}
+	}
```

Comment:
The logic in this test needs a bit of an improvement:
* The query is not set as the main query, so it won't actually run the relevant logic.
* Please also add an assertion that _before_ the loop is entered the function returns `'fetchpriority' => 'high'`.

Reply:
@mukeshpanchal27 What I meant is:
* The `$query` used here is not set as the main query, so that means the test is not actually covering that logic. You need to add the function call to set that query as the main query, similar to the test above.
* The other point is to add another assertion that an image _before_ the loop is not lazy-loaded. Right now you are only asserting that an image _in_ the loop is lazy-loaded, but the reverse is just as important to cover.

Reply:
Thank you, @felixarntz, for the explanation. I have updated the unit test code as per your instructions.